From 798c54fd565519870920888059a23ada00f64992 Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinguyen@altera.com>
Date: Fri, 7 Dec 2012 12:15:06 -0600
Subject: [PATCH 024/248] FogBugz #89246:Boot Linux on actual hardware

Boot up Linux on devkit using SDRAM on FPGA with an address
location of 0xC0000000. A kconfig option CONFIG_FPGA_SDRAM
has been introduced. This kconfig option is for booting
Linux using the SDRAM on the FPGA.

Also remove NAND support from socfpga_defconfig. Can always
be enabled by menuconfig.

NOTE: To boot Linux on VT or ICE, need to make sure that
CONFIG_FPGA_SDRAM is not selected.

[Original patch taken from
git://git.rocketboards.org/linux-socfpga.git socfpga-3.14]

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
Signed-off-by: Dinh Nguyen <dinguyen@altera.com>
---
 arch/arm/configs/socfpga_defconfig | 6 +-----
 arch/arm/mach-socfpga/Kconfig      | 6 ++++++
 2 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/arch/arm/configs/socfpga_defconfig b/arch/arm/configs/socfpga_defconfig
index 50ce2c0..19112e2 100644
--- a/arch/arm/configs/socfpga_defconfig
+++ b/arch/arm/configs/socfpga_defconfig
@@ -17,6 +17,7 @@ CONFIG_MODULE_UNLOAD=y
 # CONFIG_IOSCHED_CFQ is not set
 CONFIG_ARCH_SOCFPGA=y
 CONFIG_MACH_SOCFPGA_CYCLONE5=y
+CONFIG_FPGA_SDRAM=y
 CONFIG_ARM_THUMBEE=y
 # CONFIG_ARCH_VEXPRESS_CORTEX_A5_A9_ERRATA is not set
 # CONFIG_CACHE_L2X0 is not set
@@ -46,11 +47,6 @@ CONFIG_MTD=y
 CONFIG_MTD_CHAR=y
 CONFIG_MTD_BLOCK=y
 CONFIG_MTD_M25P80=y
-CONFIG_MTD_NAND_ECC=y
-CONFIG_MTD_NAND=y
-CONFIG_MTD_NAND_DENALI=y
-CONFIG_MTD_NAND_DENALI_DT=y
-CONFIG_MTD_NAND_IDS=y
 CONFIG_PROC_DEVICETREE=y
 CONFIG_BLK_DEV_RAM=y
 CONFIG_BLK_DEV_RAM_COUNT=2
diff --git a/arch/arm/mach-socfpga/Kconfig b/arch/arm/mach-socfpga/Kconfig
index b5c8238..10ff42f 100644
--- a/arch/arm/mach-socfpga/Kconfig
+++ b/arch/arm/mach-socfpga/Kconfig
@@ -16,3 +16,9 @@ config ARCH_SOCFPGA
 	select MIGHT_HAVE_CACHE_L2X0
 	select SPARSE_IRQ
 	select USE_OF
+
+config FPGA_SDRAM
+	bool "Enables FPGA SDRAM"
+	depends on ARCH_SOCFPGA
+	help
+		This enables the SDRAM controller that is connected to the FGPA.
-- 
1.9.1

